visualizations.py

```diff
@@ -123,13 +134,124 @@ def display_embedding_graph(model_builder):
+        amount_of_perturbation_random: Array containing amount of 
+        perturbation for randomly perturbed input.
+
+    '''
+    amount_of_perturbation_fgsm = []
+    amount_of_perturbation_random = []
+    
+
+    init_variables(model_builder, ranker, constants._TEST_DATA_PATH,
+                   answer_num, epsilons[0], ref_num, True)
+    for perturb in epsilons:
+        init_variables(
+            model_builder,
+            ranker,
+            constants._TEST_DATA_PATH,
+            answer_num,
+            perturb,
+            ref_num,
+            False)
+        change_in_rank_fgsm = abs(
+            model_builder.ranking_array[answer_num] -
+            model_builder.perturbed_ranking_array[answer_num])
+        amount_of_perturbation_fgsm.append(change_in_rank_fgsm)
+
+        change_in_rank_random = abs(
+            model_builder.ranking_array[answer_num] -
+            model_builder.random_ranking_array[answer_num])
+        amount_of_perturbation_random.append(change_in_rank_random)
+    return amount_of_perturbation_fgsm, amount_of_perturbation_random
+
+
+def perturbation_vs_epsilon_graph(model_builder, ranker, answer_num, ref_num, argv):
+    """
+    Creates and saves bar graph showing difference in perturbation for both
+    fgsm-perturbed and random-perturbed input for a variety of epsilon values.
+
+    Args:
+        model_builder: The model in use.
+        ranker: Ranking object:
+        answer_num: The answer that is perturbed.
+        ref_num: The reference answer for the perturbed answer.
+        argv: Command line input to be used for directory name.
+    """
+    #range of epsilons can be modified as desired to change
+    #number of bars or increment size
+    epsilons = np.arange(.1, .5, .1)
+    amount_of_perturbation_fgsm, amount_of_perturbation_random = perturbation_vs_epsilon(
+        model_builder, ranker, answer_num, ref_num, epsilons)
+
+    barWidth = 0.25
+    plt.figure(figsize=(22, 8))
+    plt.rcParams.update({'font.size': 22})
+
+    # Set position of bar on X axis
+    r1 = np.arange(1, len(amount_of_perturbation_fgsm) + 1)
+    r2 = [x + barWidth for x in r1]
+
+    barlist1 = plt.bar(
+        r1,
+        amount_of_perturbation_fgsm,
+        color='#3CAEA3',
+        width=barWidth,
+        edgecolor='white',
+        label='Perturbation with FGSM Noise')
+    barlist2 = plt.bar(
+        r2,
+        amount_of_perturbation_random,
+        color='#20639B',
+        width=barWidth,
+        edgecolor='white',
+        label='Perturbation with Random Noise')
+
+    # Add xticks on the middle of the group bars
+    plt.xlabel('Epsilon Value', fontweight='bold')
+    plt.xticks(
+        [r for r in range(1, len(amount_of_perturbation_fgsm) + 1)], (str(num)[:3] for num in epsilons))
+
+    plt.ylabel('Difference in Ranking Score', fontweight='bold')
+    plt.title('Amount of Perturbation vs Epsilon Value', fontweight='bold')
+
     plt.legend()
+
+    if len(argv) > 1:

```

Comment:
add an assert at the start enforcing that all required arguments are present. Also use named command line parameters, such as in the example scripts I shared with you.